client/services/utils/local_settings.js

```diff
@@ -0,0 +1,104 @@
+import EventEmitter from 'event-emitter-es6'
+
+const STORAGE_KEY = 'localsettings'
+const DATA_TYPE_KEY = 'dataType'
+const VALUE_KEY = 'value'
+const NAME_KEY = 'name'
+class LocalSettings extends EventEmitter {
+  constructor ($window, $rootScope) {
+    super()
+    Object.assign(this, { $window, $rootScope })
+    // this.STORAGE_KEY = STORAGE_KEY
+    const settingsFromSession = this.getFromLocalStorage()
+    if (settingsFromSession !== undefined) {
+      this.settings = settingsFromSession
+    } else {
+      this.settings = { }
+    }
+  }
+
+  checkSettingValidity (setting) {
+    if (setting.hasOwnProperty(NAME_KEY) && setting.hasOwnProperty(DATA_TYPE_KEY) && setting.hasOwnProperty(VALUE_KEY)) {
+      const dataType = setting[DATA_TYPE_KEY]
+      let valid = false
+      if (dataType === 'number' || dataType === 'boolean' || dataType === 'string') {
+        // eslint-disable-next-line valid-typeof
+        valid = (typeof (setting[VALUE_KEY]) === dataType)
+      }
+      return valid
+    }
+    return false
+  }
+
+  addSettings (namespace, namespaceSettingsArray) {
+    const settings = this.settings
+    if (this.settings.hasOwnProperty(namespace)) {
+      namespaceSettingsArray.forEach(setting => {
+        const settingToUpdate = settings[namespace].find(entry => entry.name === setting.name)
+        if (settingToUpdate) {
+          settingToUpdate[VALUE_KEY] = setting[VALUE_KEY]
+        } else {
+          settings[namespace].push(setting)
+          this.$rootScope.$watch(() => setting[VALUE_KEY], (newValue, oldValue) => {
+            if (newValue !== oldValue) {
+              this.emit(`${namespace} ${setting[NAME_KEY]}`)
+            }
+          })
+        }
+      })
+    } else {
+      settings[namespace] = namespaceSettingsArray
+    }
+    this.settings = settings
+    this.saveToLocalStorage()
+  }
+
+  update (passedSettings, passedNamespace) {
```

Comment:
The names `passedSettings` and `passedNamespace` are confusing. We know they have been passed, we see the function. What are they more? Are they the new settings? new namespace? or are they just the settings and namespace and there are not others?

Reply:
Is this all that is left?

Reply:
No, al the other open conversations are still relevant